### DISCUSSION TITLE
Custom Klipper Repos

### DISCUSSION THREAD

**Discussion by zarecor60** [Arksine/moonraker#463] [Q&A] [answered]
Is it possible to define custom repositories for klipper with the update manager? I did some quick digging into it, and it appears to be hard-coded. I was curious if there's a way to override the defaults for this, I have a person fork with a couple patches to the PID controls that I haven't been confident enough to push upstream yet, and am wondering if ti's possible to my repo rather than the upstream.

**Comment by Arksine** [comment, marked as answer]
Yes, see the [enable_repo_debug](https://moonraker.readthedocs.io/en/latest/configuration/#update_manager) option.  When this is enabled the update manager will bypass the validation check, letting you update from unofficial remotes.  It can also perform an update when head is detached.

**Reply by zarecor60** [reply]
Great! Thanks for the reply, that's exactly what I was looking for.
